### ISSUE TITLE
RealizeText missing an EnvironmentObject, causing a fatal error

### ISSUE THREAD

**Issue by goodones-mac** [judoapp/judo-swiftui#1]
Version 4.1:

Testing out the example BumbleOnboarding.judo file in my app, I get a fatal error for a missing DocumentLocalizations environment object in the view, and I can't add it myself from the top either.  It happens when I press the next button on the bottom right.

<img width="1411" alt="Screenshot 2023-10-30 at 2 45 49 PM" src="https://github.com/judoapp/judo-swiftui/assets/123040996/2d402b0a-1f31-4875-af4f-abb16a1ae888">

<img width="473" alt="Screenshot 2023-10-30 at 2 47 44 PM" src="https://github.com/judoapp/judo-swiftui/assets/123040996/5d89ec1d-0e36-435d-b5f7-2f82b5829c59">

```
import SwiftUI
import Judo
import JudoModel

struct JudoPlaygroundView: View {
    let stubLocal = DocumentLocalizations()
    
    var body: some View {
        Judo.View("BumbleOnboarding.judo")
            .environmentObject(stubLocal)   
    }
}

#Preview {
    JudoPlaygroundView()
}
```

[BumbleOnboarding.zip](https://github.com/judoapp/judo-swiftui/files/13210105/BumbleOnboarding.zip)

Because of this runtime crash nature of `EnvironmentObject`, I avoid it in my own project. It's better to keep dependencies checked in compile time or recoverable in runtime, which EnvironmentObject is neither.

**Comment by krzyzanowskim** [comment]
Hi, I'm trying to reproduce the crash with SDK 4.1 but am unable to reproduce it when I run it on a device or Simulator.
I have attached the minimal Xcode project. 

[Demo.zip](https://github.com/judoapp/judo-swiftui/files/13238398/Demo.zip)

Can you please run it and see if it works or crash?

**Comment by goodones-mac** [comment]
The Judo view is being pushed on via a navigation link.  The cause is having two embedded NavigationView / NavigationStack objects inside each other.   

Demo video: https://www.loom.com/share/dbb4fdf3a224458d835fe257d9031a79?sid=34b51164-afeb-4c2e-b02b-d4625e94f7a8

```swift
@main
struct DemoApp: App {
    var body: some Scene {
        WindowGroup {
            if #available(iOS 16.0, *) {
                NavigationStack {
                    ContentView()
                }
            } else {
                NavigationView {
                    ContentView()
                }
            }
        }
    }
}
```

**Comment by krzyzanowskim** [comment]
Thank you. The issue is addressed in v5, which is in public beta. SDK changes are on `main` branch for testing.
